### PR TITLE
Make Kotlin modules compatible with Kotlin 1.6

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

Kotlin modules provided by Armeria do not specify a language and API version at all. As a result, Kotlin 1.9 was set by default. That makes Armeria Kotlin modules incompatible with Kotlin version

Modifications:

- Introduced a new flag scheme for `kotlin` flag to specify target compatibility with `kotlin(\\d+\\.\\d+)`.
  - For example, `kotlin1.6` sets 1.6 to the language and API version
- Apply `kotlin1.6` flag to `armeria-kotlin` and `armeria-grpc-kotlin` modules.

Result:

Armeria Kotlin modules are now compatible with Kotlin 1.6

### CODE REVIEW

**Review by jrhee17 — APPROVED** [review]
This looks ready for review (and merge!) 👍  Thanks @ikhoon  🙇 👍 🙇

File: gradle/scripts/README.md [hunk 1]
```diff
@@ -662,6 +663,15 @@ Plus, you can use `-PminimumJavaVersion` property to override the minimum versio
 set by `java` flag. For example, if you set `-PminimumJavaVersion=11`, `javaTargetCompatibility` lower than 
 Java 11 will be upgraded to Java 11.
 
+## Setting a Kotlin target version with the `kotlin(\\d+\\.\\d+)` flag.
+
+By default, `kotlin` flag compile a Kotlin module with the [language version and API version](https://kotlinlang.org/docs/compatibility-modes.html)
```
Comment (minwoox) [inline gradle/scripts/README.md:668]:
```suggestion
By default, `kotlin` flag compiles a Kotlin module with the [language and API version](https://kotlinlang.org/docs/compatibility-modes.html)
```

**Review by minwoox — COMMENTED** [review]
Thanks! Could you send this PR to https://github.com/line/gradle-scripts?

**Review by trustin — APPROVED** [review]
:bow::+1:

**Comment by ikhoon** [comment]
The changes in line/scripts has been updated to #5149

**Pull request closed without merging**